### PR TITLE
Port Selenium Java `get text` tests to webdriver

### PR DESCRIPTION
**Pull request by shs96c**
The `Element Get Text` command says that implementations
must exactly match Selenium's existing text handling. To
ensure this, the tests from Selenium's atoms and webdriver
implementations are being ported to the WPT suite.

This PR is equivalent to the tests in
//java/client/test/org/openqa/selenium:TextHandlingTest

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/6746)

### CODE REVIEW

**Comment by shs96c** [comment]
Although I've done my best to remove duplication, there may be some overlap between these tests and the ones in #6730. I think it best to get the tests into the tree, and then do a second pass better organising them into something easier for a normal person to navigate.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260741598)

# Firefox (nightly) #
Testing web-platform-tests at revision 1cf37631325f0fbcea3e8d57b8c1463a9f7566be
Using binary /home/travis/build/w3c/web-platform-tests/_venv/firefox/firefox
Starting tests
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6746/webdriver/tests/state/text/text_handling_test.py">/webdriver/tests/state/text/text_handling_test.py</a></summary>


|                                                      Subtest                                                       | Results | Messages |
|--------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                    | OK      |          |
| `text_handling_test.py::test_ignore_script_tags`                                                                   | PASS    |          |
| `text_handling_test.py::test_text_should_be_trimmed`                                                               | PASS    |          |
| `text_handling_test.py::test_does_not_collapse_non_breaking_spaces`                                                | PASS    |          |
| `text_handling_test.py::test_does_not_trim_non_breaking_spaces_at_the_start_of_a_line_in_the_middle_of_text`       | PASS    |          |
| `text_handling_test.py::test_does_not_trim_whitespace_when_line_wraps`                                             | PASS    |          |
| `text_handling_test.py::test_whitespace_in_inline_elements`                                                        | PASS    |          |
| `text_handling_test.py::test_only_include_visible_text`                                                            | PASS    |          |
| `text_handling_test.py::test_text_of_an_input_element_should_be_empty`                                             | PASS    |          |
| `text_handling_test.py::test_text_of_a_textarea_should_be_equal_to_its_default_text`                               | PASS    |          |
| `text_handling_test.py::test_text_of_a_textarea_should_be_equal_to_its_default_text_even_after_typing`             | PASS    |          |
| `text_handling_test.py::test_text_of_a_textarea_should_be_equal_to_its_default_text_even_after_changing_the_value` | PASS    |          |
| `text_handling_test.py::test_can_get_text_that_is_a_valid_json_object`                                             | PASS    |          |
| `text_handling_test.py::test_handles_text_that_looks_like_a_numer`                                                 | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260741599)

# Sauce (safari) #
Testing web-platform-tests at revision 1cf37631325f0fbcea3e8d57b8c1463a9f7566be
Using binary None
Starting tests

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260741607)

# Chrome (unstable) #
Testing web-platform-tests at revision 1cf37631325f0fbcea3e8d57b8c1463a9f7566be
Using binary None
Starting tests
No tests run.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260741608)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 1cf37631325f0fbcea3e8d57b8c1463a9f7566be
Using binary None
Starting tests
No tests run.

**Comment by jgraham** [comment]
So, I hate to be that person, but is this OK license-wise?

**Comment by shs96c** [comment]
Two responses:

1/ Aren't the Apache 2 and W3C licenses compatible?
2/ Even if they're not, this is a rewrite of all the tests using a different language and infra --- isn't this different enough to count as new work? 

I can raise this with the SFC if that'd be useful.

**Comment by jgraham** [comment]
I don't think they're compatible in the sense that you can randomly move code under one license to a different license. And I think that a port is explicitly a derived work, and so normal copyright provisions apply. But am am not an expert.

**Comment by shs96c** [comment]
After talking with the SFC's General Counsel:

> Under Apache 2.0, you can redistribute modified versions of
> Selenium files (including files ported into a different programming
> language) under a different license, including the W3C's 3-clause BSD
> license.  Be sure to add a note indicating that the files have been
> modified, and include the appropriate copyright/attribution notices (and
> patent/trademark, if applicable) for the original files.

So I think we're good. I can add the missing copyright and attribution notice.

File: webdriver/tests/state/text/text_handling_test.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import pytest
+
+from tests.support.inline import inline
+
```
Comment (AutomatedTester) [inline webdriver/tests/state/text/text_handling_test.py]: Can we get the other steps for this command added.

Comment (shs96c) [inline webdriver/tests/state/text/text_handling_test.py]: This is just a straight port of the selenium tests. I want them in their own files as we need to give attribution to them.

File: webdriver/tests/state/text/text_handling_test.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    session.execute_script("arguments[0].value = arguments[1]", args=[element.json(), "New Text"])
+    assert element.text == "Example text"
+
+def test_can_get_text_that_is_a_valid_json_object(session):
+    session.url = inline("""<span id="simpleJsonText">{a="b", c=1, d=true}</span>""")
+    element = session.find.css("#simpleJsonText", all=False)
+    assert element.text == "{a=\"b\", c=1, d=true}"
+
+def test_handles_text_that_looks_like_a_numer(session):
+    session.url = inline("""<div id='point'>12.345</div><div id='comma'>12,345</div><div id='space'>12 345</div>""")
```
Comment (shs96c) [inline webdriver/tests/state/text/text_handling_test.py]: These tests are particularly sensitive to whitespace. I'll clean up the ones that I'm certain are safe.

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None


<h2>Failing Jobs</h2>
<ul>

<li>firefox:nightly</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17661)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/267205062)

**Comment by jgraham** [comment]
Is there a plan to merge these tests, or should we close this PR?

**Comment by stephenmcgruer** [comment]
Closing given no reply to the previous request in Dec 2019. Please re-open if there is still a desire to merge these (@shs96c @AutomatedTester )

**Pull request closed without merging**